spinn_front_end_common/interface/java_caller.py

```diff
@@ -0,0 +1,158 @@
+            https://github.com/SpiNNakerManchester/JavaSpiNNaker. \
+            It must also have been built! \
+            If None the assumption is that it is the same parent directory as \
+            https://github.com/SpiNNakerManchester/SpiNNFrontEndCommon.
+        :raise ConfigurationException if simple parameter checking fails.
+        """
+        self._json_folder = json_folder
+
+        self._java_call = java_call
+        result = subprocess.call([self._java_call, '-version'])
+        if result != 0:
+            raise ConfigurationException(
+                " {} -version failed. "
+                "Please set [Java] java_call to the absolute path "
+                "to start java. (in config file)".format(self._java_call))
+
+        if javaspinnaker_path is None:
+            interface = os.path.dirname(os.path.realpath(__file__))
+            spinn_front_end_common = os.path.dirname(interface)
+            spinnfrontendcommon = os.path.dirname(spinn_front_end_common)
+            parent = os.path.dirname(spinnfrontendcommon)
+            self._javaspinnaker_path = os.path.join(parent, "JavaSpiNNaker")
+        else:
+            self._javaspinnaker_path = javaspinnaker_path
+        if not os.path.isdir(self._javaspinnaker_path):
+            raise ConfigurationException(
+                "No Java code found at {}".format(self._javaspinnaker_path))
+
+        self._machine_json = None
+        self._placement_json = None
+
+    def set_machine(self, machine):
+        """
+        Passes the machine in leaving this class to decide pass it to Java.
+
+        :param machine: A machine Object
+        """
+
+        algo = ConvertToJavaMachine()
+        path = os.path.join(self._json_folder, "machine.json")
+        self._machine_json = algo(machine, path)
+
+    def set_database(self, database_file):
+        """
+        Passes the database file in.
+
+        :param database_file: Path to the sql lite databse
+        :type database_file: str
+        """
+        self._database_file = database_file
+
+    def set_placements(self, placements, transceiver):
+        """
+        Passes in the placements leaving this class to decide pass it to Java.
+
+        This method may obtain extra information about he placements which is \
+            why it also needs the transceiver.
+
+        Currently the extra information extracted is recording region
+            base address but this could change if recording region saved in
+            the database.
+
+        Currently this method uses json but that may well change to using the
+            database.
+
+        :param placements: The Placements Object
+        :param transceiver: The Transceiver
+        """
+        path = os.path.join(self._json_folder, "java_placements.json")
+        self._placement_json = self._write_placements(
+            placements, transceiver, path)
+
+    def _write_placements(self, placements, transceiver, path):
```

Comment:
```suggestion
    
    @staticmethod
    def _write_placements(placements, transceiver, path):
```

Reply:
what does it matter for an underscore method.  For a non underscore I could agree.

Reply:
Even more as this method is probably replaced in extractor2

Reply:
coz pycharm complained that it should be static, as it doesnt use anything from self. reducing warnings etc. are good in prinicple

Reply:
done